alchemist/alchemist-incarnation-biochemistry/src/main/java/it/unibo/alchemist/model/implementations/nodes/CellNodeImpl.java

```diff
@@ -15,29 +15,51 @@
 import java.util.Map;
 import java.util.Set;
 
+import org.apache.commons.math3.util.FastMath;
+
 import com.google.common.collect.MapMaker;
 
 import it.unibo.alchemist.model.implementations.molecules.Biomolecule;
 import it.unibo.alchemist.model.implementations.molecules.Junction;
+import it.unibo.alchemist.model.implementations.positions.Continuous2DEuclidean;
+import it.unibo.alchemist.model.interfaces.CellWithCircularArea;
 import it.unibo.alchemist.model.interfaces.Environment;
-import it.unibo.alchemist.model.interfaces.ICellNode;
+import it.unibo.alchemist.model.interfaces.CellNode;
 import it.unibo.alchemist.model.interfaces.Molecule;
+import it.unibo.alchemist.model.interfaces.Position;
 
 /**
  *
  */
-public class CellNode extends DoubleNode implements ICellNode {
+public class CellNodeImpl extends DoubleNode implements CellNode, CellWithCircularArea {
 
     private static final long serialVersionUID = 837704874534888283L;
 
-    private final Map<Junction, Map<ICellNode, Integer>> junctions = new MapMaker().concurrencyLevel(2).makeMap();
+    private final Map<Junction, Map<CellNode, Integer>> junctions = new MapMaker()
+            .concurrencyLevel(2).makeMap();
```

Comment:
Sorry, This part of code's not mine, and I don't know anything about concurrencyLevel. Should I change that with a simple HashMap?